scala/private/phases/api.bzl

```diff
@@ -22,18 +22,19 @@ def _adjust_phases(phases, adjustments):
     # phase_name: the name of the new phase, also used to access phase information
     # phase_function: the function of the new phase
     for (relation, peer_name, phase_name, phase_function) in adjustments:
-        for idx, (needle, _) in enumerate(phases):
-            if relation in ["^", "first"]:
-                phases.insert(0, (phase_name, phase_function))
-            elif relation in ["$", "last"]:
-                phases.append((phase_name, phase_function))
-            elif needle == peer_name:
-                if relation in ["-", "before"]:
-                    phases.insert(idx, (phase_name, phase_function))
-                elif relation in ["+", "after"]:
-                    phases.insert(idx + 1, (phase_name, phase_function))
-                elif relation in ["=", "replace"]:
-                    phases[idx] = (phase_name, phase_function)
+        if relation in ["^", "first"]:
```

Comment:
Even though I still can't think of a failing case of original logic, the fix looks more reasonable. Thanks for catching it!